homeassistant/components/axis/config_flow.py

```diff
@@ -154,14 +162,22 @@ async def async_step_reauth(
             CONF_NAME: entry_data[CONF_NAME],
             CONF_HOST: entry_data[CONF_HOST],
         }
+        return await self._redo_configuration(entry_data, keep_password=False)
 
+    async def _redo_configuration(
+        self, entry_data: Mapping[str, Any], keep_password: bool
+    ) -> ConfigFlowResult:
+        """Re-run configuration step."""
         self.discovery_schema = {
             vol.Required(
-                CONF_PROTOCOL, default=entry_data.get(CONF_PROTOCOL, DEFAULT_PROTOCOL)
+                CONF_PROTOCOL, default=entry_data.get(CONF_PROTOCOL, "http")
```

Comment:
It looks like the fallback changed from `https` to `http` here. Is that what you wanted?

Reply:
Yes, as I introduced that parameter just the other day, I realised when doing the tests here as the port would still be whatever configured before and that would mean "http" protocol I changed back to http in this specific case. Once its been populated it should always be known what the user has configured.

Reply:
I'm not sure why you just wouldn't change the default in that case.

If the user is reconfiguring and hasn't previously set the protocol, the default is http, but in all other cases the default is https?

Probably I am missing something.

Reply:
Well the port must follow the protocol right, and I don't want to meddle too much in existing configurations if they explicitly want to stay on http